espnetez/trainer.py

```diff
@@ -185,6 +217,64 @@ def __init__(
         build_model_fn=None,
         **kwargs
     ):
+        """Initialize an EZ training environment.
+
+        Args:
+            task (str): Identifier of the EZ task to be trained.
+            train_config (dict|argparse.Namespace): Configuration for the training
+                run. If adictionary, the key/value pairs are converted into an
+                ``argparse.Namespace``. Any additional keyword arguments passed via
+                ``**kwargs`` are merged into ``train_config``.
+            output_dir (str|pathlib.Path): Directory where the trained model and other
+                artifacts will be written.
+            stats_dir (str|pathlib.Path): Directory where training statistics and logs
+                should be stored.
+            data_info (dict, optional): Metadata describing the training and validation
+                datasets.  The structure can be either:
+                ``{"train": {...}, "valid": {...}}`` or a flat mapping where the same
+                items are used for both splits.  Each value must be a tuple
+                ``(file_name, name, type)``.
```

Comment:
I think it's good to link where we can find the types.
https://github.com/espnet/espnet/blob/eaf9a83f5f5a7427651b2eb4b6438a9a6427ebf0/espnet2/train/dataset.py#L248